crates/test-collector/src/lib.rs

```diff
@@ -63,10 +64,30 @@ pub enum ExpectedTestResult {
 
 #[derive(Debug, Clone, PartialEq)]
 pub enum ForkConfig {
+    Id(String),
+    Params(Url, BlockId),
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum RawForkConfig {
     Id(String),
     Params(String, BlockId),
 }
 
+fn fork_config_from_raw_fork_config(
+    maybe_raw_fork_config: Option<RawForkConfig>,
+) -> Result<Option<ForkConfig>> {
+    let result = match maybe_raw_fork_config {
+        Some(RawForkConfig::Params(raw_url, block_id)) => {
+            let url = Url::parse(&raw_url)?;
+            Some(ForkConfig::Params(url, block_id))
+        }
+        Some(RawForkConfig::Id(name)) => Some(ForkConfig::Id(name)),
```

Comment:
Why do we need to leak ID logic to the deeper parts of the codebase instead of resolving it somewhere here?

Reply:
I would need to pass the Scarb config to the compilation crate, idk if that 's a good idea, but can do that